File: Java/EjercicioHerencia/src/ejercicioherencia/Persona.java
package ejercicioherencia;

class Persona {
    
    //atributos de clase
    static private int contadorPersonas;
    
    //atributos de instancia
    private int idPersona;
    private String nombre;
    private String apellido;
    private int edad;
    
    //constructor

    public Persona(String nombre, String apellido, int edad) {
        this.nombre = nombre;
        this.apellido = apellido;
        this.edad = edad;
        this.idPersona = ++Persona.contadorPersonas;
    }
    
    //getters and setters
    public int getIdPersona() {
        return this.idPersona;
    }

    public String getNombre() {
        return this.nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido() {
        return this.apellido;
    }
    
    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    public int getEdad() {
        return this.edad;
    }

    public void setEdad(int edad) {
        this.edad = edad;
    }
    
    //métodos
    @Override
    public String toString(){
        return this.getIdPersona() + " "
             + this.getNombre() + " "
             + this.getApellido() + " " 
             + this.getEdad();
    }
}

File: UJS2021-Udemy/Sección11/Persona.js
class Persona {
     
    static contadorPersonas = 0;

    constructor(nombre, apellido, edad) {
        this.idPersona = ++Persona.contadorPersonas;
        this.nombre = nombre; 
        this.apellido = apellido;
        this.edad = edad;
    }

    //getters
    get getNombre() {
        return this.nombre;
    }
    get getApellido() {
        return this.apellido;
    }
    get getEdad() {
        return this.edad;
    }
    get getIdPersona() {
        return this.getIdPersona;
    }

    //setters
    /**
     * @param {String} nombre
     */
    set setNombre(nombre) {
        this.nombre = nombre;
    }
    /**
     * @param {String} apellido
     */
    set setApellido(apellido) {
        this.apellido = apellido;
    }
    /**
     * @param {Number} edad
     */
    set setEdad(edad) {
        if(edad > 0) {
            this.edad = edad;
        }
    }

    //métodos
    //Overriding
    toString() {
        return `id=${this.idPersona}: ${this.nombre} ${this.apellido}, ${this.edad} años`;
        //return 'id=' + this.idPersona +': '+ this.nombre +' '+ this.apellido +', '+ this.edad + ' años';
        //usamos el concepto de template string
       
    }
}